### PR TITLE
improved documentation of clang-format usage

### PR DESCRIPTION
**Pull request by BrianMarre**
The current documentation does not mention that clangFormat is not included with git, but must be installed/loaded in addition.

In addition I added a short example workflow based on my short introduction by S.Bastrakov.

### CODE REVIEW

File: docs/source/dev/styleguide.rst [hunk 1]
```diff
@@ -31,22 +31,53 @@ To format all files in your working copy, you can run this command in bash from
...
 
-Format Changes Using Git
-^^^^^^^^^^^^^^^^^^^^^^^^
+Format Only Changes, Using Git
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Instead of using the bash command above you can use *Git* together with *ClangFormat* to format your patched code only.
+
+    *ClangFormat* is an external tool for code formating that can be called by *Git* on changed files only and
+    is included in the clang library.
```
Comment (psychocoderHPC) [inline docs/source/dev/styleguide.rst:40]: `... and is part of clang tools`

clang -format is part of clang tools, not the clang library https://clang.llvm.org/docs/ClangTools.html

File: docs/source/dev/styleguide.rst [hunk 2]
```diff
@@ -31,22 +31,53 @@ To format all files in your working copy, you can run this command in bash from
...
+
+.. code-block:: bash
+
+   git clangFormat
+
 .. warning::
 
-    The binary for *ClangFormat* is on some operating systems called `clang-format`.
-    If so please check that `clang-format --version` returns version `11.X.X`.
+    The binary for *ClangFormat* is instead called `clang-format` on some operating systems.
```
Comment (psychocoderHPC) [inline docs/source/dev/styleguide.rst:56]: I would remove the word `instead`

File: docs/source/dev/styleguide.rst [hunk 3]
```diff
@@ -31,22 +31,53 @@ To format all files in your working copy, you can run this command in bash from
...
+
+3. format them according to guidelines
+.. code-block:: bash
+
+    git clangFormat
+
+4. stage the now changed(formated) files again
+.. code-block:: bash
+
+    git add <files you changed>
```
Comment (psychocoderHPC) [inline docs/source/dev/styleguide.rst:75]: you need an empty line to close the code block

File: docs/source/dev/styleguide.rst [hunk 4]
```diff
@@ -31,22 +31,53 @@ To format all files in your working copy, you can run this command in bash from
...
-    The binary for *ClangFormat* is on some operating systems called `clang-format`.
-    If so please check that `clang-format --version` returns version `11.X.X`.
+    The binary for *ClangFormat* is instead called `clang-format` on some operating systems.
+    If *clangFormat* is not recognized try *clang-format* instead, in addition please check that `clang-format --version` returns version `11.X.X` in this case.
+
+The Typical workflow using git clangFormat is the following,
+
+1. make your patch
+2. stage the changed files in git
+.. code-block:: bash
```
Comment (psychocoderHPC) [inline docs/source/dev/styleguide.rst:63]: a block requires an empty line before the command.

You can preview your PR if you scroll down to the tests, one test is building the documentation.
For the current commit the documentation can be found: https://picongpu--3522.org.readthedocs.build/en/3522/dev/styleguide.html?highlight=clang%20format#format-only-changes-using-git

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]

**Comment by psychocoderHPC** [comment]
note: I disabled compile tests for this PR to avoid that we block the CI.

File: docs/source/dev/styleguide.rst (hunk 1 above)
Comment (BrianMarre) [reply docs/source/dev/styleguide.rst]: fixed

File: docs/source/dev/styleguide.rst (hunk 2 above)
Comment (BrianMarre) [reply docs/source/dev/styleguide.rst]: agree, done

File: docs/source/dev/styleguide.rst (hunk 3 above)
Comment (BrianMarre) [reply docs/source/dev/styleguide.rst:79]: fixed

File: docs/source/dev/styleguide.rst (hunk 4 above)
Comment (BrianMarre) [reply docs/source/dev/styleguide.rst:65]: fixed, and thanks for the pointer

**Comment by psychocoderHPC** [comment]
could you please run on your system:

```
git commit --amend -a
```

And force push the branch again. The command will change your commit hash and the force push will trigger the CI again. For unknown reason the CI was not reporting the status back.

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**